File: inject-java/src/test/groovy/io/micronaut/aop/introduction/with_around/ObservableInterceptor.java
package io.micronaut.aop.introduction.with_around;

import io.micronaut.aop.MethodInterceptor;
import io.micronaut.aop.MethodInvocationContext;

import io.micronaut.core.annotation.Nullable;
import jakarta.inject.Singleton;

@Singleton
public class ObservableInterceptor implements MethodInterceptor<Object, Object> {
    @Nullable
    @Override
    public Object intercept(MethodInvocationContext<Object, Object> context) {
        return "World";
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/aop/introduction/with_around/ObservableInterceptor.kt
package io.micronaut.kotlin.processing.aop.introduction.with_around

import io.micronaut.aop.MethodInterceptor
import io.micronaut.aop.MethodInvocationContext
import jakarta.inject.Singleton

@Singleton
class ObservableInterceptor : MethodInterceptor<Any?, Any?> {

    override fun intercept(context: MethodInvocationContext<Any?, Any?>?): Any {
        return "World"
    }
}
